server/routes/summary_routes.ts

```diff
@@ -128,15 +114,25 @@ export function registerSummaryAssistantRoutes(
         return res.ok({ body: response.body.inference_results[0].output[0].result });
       } catch (e) {
         return res.internalError();
-      } finally {
-        // Reset both agents id in case of update
-        userInsightAgentId = undefined;
-        osInsightAgentId = undefined;
       }
     })
   );
 }
 
+function detectInsightAgentId(
+  insightType: string,
+  summaryType: string,
+  client: OpenSearchClient['transport']
+) {
+  // We have separate agent for os_insight and user_insight. And for user_insight, we can
+  // only get it by searching on name since it is not stored in agent config.
+  if (insightType === 'os_insight') {
+    return getAgentIdByConfigName(OS_INSIGHT_AGENT_CONFIG_ID, client);
+  } else if (insightType === 'user_insight' && summaryType === 'alerts') {
+    return searchAgent({ name: 'KB_For_Alert_Insight' }, client);
+  }
```

Comment:
missing return value if `insightType` is not `os_insight` or `user_insight`

Reply:
Seems it will return undefined implicitly in that case(I tell that by seeing the return type in my IDE). But it's still better to return undefined explicitly.